plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -528,29 +537,54 @@ export class IngestionConsumer {
                 continue
             }
 
-            const eventKey = `${event.token}:${event.distinct_id}`
-
             if (this.shouldSkipPerson(event.token, event.distinct_id)) {
                 event.properties = {
                     ...(event.properties ?? {}),
                     $process_person_profile: false,
                 }
             }
 
+            batch.push({ message, event })
+        }
+
+        return Promise.resolve(batch)
+    }
+
+    private groupEventsByDistinctId(messages: IncomingEventWithTeam[]) {
+        const batches: IncomingEventsByDistinctId = {}
+        for (const { event, message, team, token } of messages) {
+            const eventKey = `${token ?? ''}:${event.distinct_id}`
+
             // We collect the events grouped by token and distinct_id so that we can process batches in parallel whilst keeping the order of events
             // for a given distinct_id
             if (!batches[eventKey]) {
                 batches[eventKey] = {
-                    token: event.token ?? '',
+                    token: token ?? '',
                     distinctId: event.distinct_id ?? '',
                     events: [],
                 }
             }
 
-            batches[eventKey].events.push({ message, event })
+            batches[eventKey].events.push({ message, event, team, token })
         }
+        return batches
+    }
 
-        return Promise.resolve(batches)
+    private async resolveTeams(messages: IncomingEvent[]): Promise<IncomingEventWithTeam[]> {
+        const resolvedMessaged: IncomingEventWithTeam[] = []
+        for (const { event, message } of messages) {
+            const result = await populateTeamDataStep(this.hub, event)
```

Comment:
issue: looking at the `populateTeamDataStep` tests, this throws on invalid tokens, so it's going to crash the process, which we don't want.

We should stop throwing from `populateTeamDataStep`, as it's hard to catch a generic error and it's not an unexpected case, and return a null result.

Reply:
Fixed